doc/debugging.rst

```diff
@@ -0,0 +1,76 @@
+Debugging
+^^^^^^^^^
+
+All controllers and hardware interfaces are plugins that are loaded into the ``controller_manager``. Therefore the debugger
+needs to be attached to the controller_manager
+
+How-To
+******************
+
+* Install xterm, gdb and gdbserver on your system
+
+  .. code-block:: bash
+
+    sudo apt install xterm gdb gdbserver
+* Make sure you run a debug/release with debug information build: 
+  This is done by passing ``--cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo`` to colcon build.
+  Remember that in Release builds some breakpoints might not behave as you think as the the corresponding line might have been removed by the optimizer. For certain cases a full Debug build (``--cmake-args -DCMAKE_BUILD_TYPE=Debug``)
+
+* Adapt the launch file to run the controller manager with the debugger attached:
+
+  * Version A: Run it directly with the gdb cli:
+
+    Add ``prefix=['xterm -e gdb -ex run --args']`` to the ``controller_manager`` node entry in your launch file.
+    Due to the way ros2launch works we need to run the specific node in a separate terminal instance. 
+    See a gdb cli tutorial for further steps
```

Comment:
A reference link would help here

Reply:
I actually googled for some. There is unfortunately no official tutorial and I didn't want to link to any third party sides that are likely to change at some point. If you have a recommendation for a gdb cli tutorial feel free to add the link

Reply:
I added a link in the references section